### PR TITLE
Add SQS contract tests

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ffc-demo-payment-service",
-  "version": "2.4.0",
+  "version": "2.4.1",
```
[the remark this answers is not in the document]
Comment (johnwatson484) [reply package.json:3]: I'm not sure I agree.  In terms of the application there are no new features added, we've just added additional types of tests to existing features.

**Comment by johnwatson484** [comment]
> What you've added looks good, however there's no new Pact (json file), so I don't understand what it's verifying?

As this is the consumer for this SQS test, as opposed to the provider as it was for the API test, Pacts are published to the `test_output` folder so they can be shared with the providers.  In this case the claim service.  Will also be the calculation service when I write the tests for the `payment` queue.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaucourt) [reply package.json:3]: Yeah, it's not massively important. My reasoning was that patch version increments refer only to bug fixes, which this isn't. Minor version increments is for private improvements:  "[Minor version] MAY be incremented if substantial new functionality or improvements are introduced within the private code."

However, I'm not prepared to 'die in a ditch' over it, it's just a matter of interpretation.

**Comment by jaucourt** [comment]
> > What you've added looks good, however there's no new Pact (json file), so I don't understand what it's verifying?
> 
> As this is the consumer for this SQS test, as opposed to the provider as it was for the API test, Pacts are published to the `test_output` folder so they can be shared with the providers. In this case the claim service. Will also be the calculation service when I write the tests for the `payment` queue.

I didn't see any new interactions added, but presumably it's different due to these being SQS tests rather than HTTP?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnwatson484) [reply package.json:3]: Ok you've convinced me.

**Comment by johnwatson484** [comment]
> > > What you've added looks good, however there's no new Pact (json file), so I don't understand what it's verifying?
> > 
> > 
> > As this is the consumer for this SQS test, as opposed to the provider as it was for the API test, Pacts are published to the `test_output` folder so they can be shared with the providers. In this case the claim service. Will also be the calculation service when I write the tests for the `payment` queue.
> 
> I didn't see any new interactions added, but presumably it's different due to these being SQS tests rather than HTTP?

Yeah they don't have interactions as such.  Well they do they're just not called interactions.  They're specified in the tests as with Http but you set the message broker up with a message definition to send and becomes the basis of the contract.

**Pull request merged**